includes/Core/Util/Method_Proxy_Trait.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\Method_Proxy_Trait
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+trait Method_Proxy_Trait {
```

Comment:
One drawback of using this is that some IDEs won't recognize the usage, and that methods that are only referenced by this proxy will appear unused.

![image](https://user-images.githubusercontent.com/1621608/94039525-82efcd00-fdd0-11ea-91ab-e22733341f93.png)
![image](https://user-images.githubusercontent.com/1621608/94039559-8e42f880-fdd0-11ea-88ea-4135cec2944d.png)

Interestingly, my IDE does recognize it when the method name is called by an anonymous function, even if it doesn't actually call the method 😄 

```php
$p = function ( $method ) {};
add_filter( 'allowed_redirect_hosts', $p( 'allowed_redirect_hosts' ) ); // works
```

This is a bit of a bummer. While we shouldn't necessarily code for our tools, this is the tradeoff we make which does somewhat negatively impact the developer experience, particularly for newcomers to the codebase. The above variant does work but feels a bit like a hack.

@felixarntz should we perhaps revert the method proxying back to the way it was without a proxy/helper or is this an acceptable tradeoff for you?

Reply:
I think that's okay for now - I prefer the simplicity of the new method wrapper. What would be nice is if PHP had something like `$this->get_oauth_client::func` or whatever :)